src/workerd/api/streams/standard.c++

```diff
@@ -1224,8 +1224,11 @@ struct ValueReadable final: public api::ValueQueue::ConsumerImpl::StateListener,
     // Here, we rely on the controller implementing the correct behavior since it owns
     // the queue that knows about all of the attached consumers.
     KJ_IF_MAYBE(s, state) {
-      auto releaseMe = kj::mv(*s);
-      return releaseMe.cancel(js, kj::mv(maybeReason));
+      KJ_DEFER({
+        auto released KJ_UNUSED = kj::mv(*s);
+        state = nullptr;
```

Comment:
Ah, took me a while to realize why `kj::mv(*s)` wouldn't work on its own -- good catch.  I assume that it is still relatively ineffectual, though...  Would `state = nullptr;` be sufficient by itself, in the defer block?

Reply:
Ahhhh, yeah, I'm unsure if it would pass muster, but some form of "consume" use of `kj::Maybe` would be nice. Technically, I believe it will work if you do `KJ_IF_MAYBE(s, kj::mv(state))`, but that would not feel as obvious as `KJ_IF_MAYBE(s, state.consume())` or `KJ_IF_MAYBE_MOVE(s, state)`.